src/django/pfb_analysis/models.py

```diff
@@ -25,6 +35,50 @@ def get_neighborhood_file_upload_path(instance, filename):
     return 'neighborhood_boundaries/{0}/{1}'.format(instance.name, os.path.basename(filename))
 
 
+class NeighborhoodManager(models.Manager):
+
+    def create_from_geom(self, geom, **kwargs):
```

Comment:
Given the comment above about overwriting the boundary each time a neighborhood shows up in the batch shapefile, this should probably be a `set_boundary_file` method on the model.